cmd/query/app/flags.go

```diff
@@ -147,6 +150,50 @@ func (qOpts *QueryOptions) BuildQueryServiceOptions(storageFactory storage.BaseF
 	return opts
 }
 
+func (qOpts *QueryOptions) BuildV2QueryServiceOptions(storageFactory storage.BaseFactory, logger *zap.Logger) *v2querysvc.QueryServiceOptions {
+	opts := &v2querysvc.QueryServiceOptions{}
+	if !initializeV2ArchiveStorage(storageFactory, opts, logger) {
+		logger.Info("Archive storage not initialized")
+	}
+
+	opts.Adjuster = v2adjuster.Sequence(v2adjuster.StandardAdjusters(qOpts.MaxClockSkewAdjust)...)
+
+	return opts
+}
+
+func initializeV2ArchiveStorage(
```

Comment:
@yurishkuro we're repeating this logic two times at the moment. Should we move this into a helper somewhere?

Reply:
ok to leave for now. We haven't finished archive factory PR, but the idea was that in the v2 extension we don't be using v1adapter but get v2 reader/writer directly from storage extension.